From f374de8f80d9d34fdc0dcd4ff3637765cb55501a Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Mon, 25 Feb 2013 23:23:16 +0100
Subject: [PATCH 06/10] Automatic conversion from DistributionImplementation
 into Copula

---
 python/src/Copula.i |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/python/src/Copula.i b/python/src/Copula.i
index b0a27e3..4285eb6 100644
--- a/python/src/Copula.i
+++ b/python/src/Copula.i
@@ -74,16 +74,21 @@ namespace OT {
     OT::CopulaImplementation * p_impl = reinterpret_cast< OT::CopulaImplementation * >( ptr );
     $1 = new OT::Copula( *p_impl );
   } else if (SWIG_IsOK(SWIG_ConvertPtr($input, &ptr, SWIG_TypeQuery("OT::Pointer<OT::DistributionImplementation> *"), 0))) {
-    // From Pointer<Implementation>
+    // From Pointer<DistributionImplementation>
     OT::Pointer<OT::DistributionImplementation> * p_impl = reinterpret_cast< OT::Pointer<OT::DistributionImplementation> * >( ptr );
     $1 = new OT::Copula( **p_impl );
+  } else if (SWIG_IsOK(SWIG_ConvertPtr($input, &ptr, SWIGTYPE_p_OT__DistributionImplementation, 0))) {
+    // From DistributionImplementation
+    OT::DistributionImplementation * p_impl = reinterpret_cast< OT::DistributionImplementation * >( ptr );
+    $1 = new OT::Copula( *p_impl );
   }
 }
 
 %typemap(typecheck,precedence=SWIG_TYPECHECK_POINTER) const OT::Copula & {
   $1 = SWIG_IsOK(SWIG_ConvertPtr($input, NULL, $1_descriptor, 0))
     || SWIG_IsOK(SWIG_ConvertPtr($input, NULL, SWIGTYPE_p_OT__CopulaImplementation, 0))
-    || SWIG_IsOK(SWIG_ConvertPtr($input, NULL, SWIG_TypeQuery("OT::Pointer<OT::DistributionImplementation> *"), 0));
+    || SWIG_IsOK(SWIG_ConvertPtr($input, NULL, SWIG_TypeQuery("OT::Pointer<OT::DistributionImplementation> *"), 0))
+    || SWIG_IsOK(SWIG_ConvertPtr($input, NULL, SWIGTYPE_p_OT__DistributionImplementation, 0));
 }
 
 %include Copula.hxx
-- 
1.7.10.4

